Services/RelationService.cs

```diff
@@ -0,0 +1,179 @@
+        }
+
+        //TODO write an algorithm for applying postal mask
+        public string ApplyPostalCodeMask(string postalCode, string postalCodeFormat)
+        {
+            return postalCode;
+        }
+
+        public IMapper ConfigureMapperForDto()
+        {
+            var mapper = new MapperConfiguration(cfg => cfg.CreateMap<Relation, RelationDTO>()
+               .ForMember("Street", opt => opt.MapFrom(c => c.RelationAddress.Street))
+               .ForMember("City", opt => opt.MapFrom(c => c.RelationAddress.City))
+               .ForMember("Country", opt => opt.MapFrom(c => c.RelationAddress.CountryName))
+               .ForMember("PostalCode", opt => opt.MapFrom(c => c.RelationAddress.PostalCode))
+               .ForMember("StreetNumber", opt => opt.MapFrom(c => c.RelationAddress.Number))).CreateMapper();
+
+            return mapper;
+        }
+
+        public IMapper ConfigureMapperForViewModel()
+        {
+            var mapper = new MapperConfiguration(cfg => cfg.CreateMap<Relation, RelationViewModel>()
+               .ForMember("Street", opt => opt.MapFrom(c => c.RelationAddress.Street))
+               .ForMember("City", opt => opt.MapFrom(c => c.RelationAddress.City))
+               .ForMember("Country", opt => opt.MapFrom(c => c.RelationAddress.CountryName))
+               .ForMember("PostalCode", opt => opt.MapFrom(c => c.RelationAddress.PostalCode))
+               .ForMember("StreetNumber", opt => opt.MapFrom(c => c.RelationAddress.Number))).CreateMapper();
+
+            return mapper;
+        }
+
+        public IQueryable<Relation> GetSortedRelationsByProperties(IQueryable<Relation> list, string orderString)
+        {
+            switch (orderString.ToUpper())
+            {
+                case "NAME":
+                    return list.OrderBy(r => r.Name);
+                case "FULLNAME":
+                    return list.OrderBy(r => r.FullName);
+                case "TELEPHONENUMBER":
+                    return list.OrderBy(r => r.TelephoneNumber);
+                case "EMAIL":
+                    return list.OrderBy(r => r.EmailAddress);
+                case "COUNTRY":
+                    return list.OrderBy(r => r.RelationAddress.Country.Name);
+                case "CITY":
+                    return list.OrderBy(r => r.RelationAddress.City);
+                case "STREET":
+                    return list.OrderBy(r => r.RelationAddress.Street);
+                case "POSTALCODE":
+                    return list.OrderBy(r => r.RelationAddress.PostalCode);
+                case "STREETNUMBER":
+                    return list.OrderBy(r => r.RelationAddress.Number);
+            }
+            return list.OrderBy(r => r.Id);
+        }
+
+        public IQueryable<Relation> GetRelationsByCategories(IQueryable<Relation> list, string filterCategory)
+        {
+            list = list.Where(c => c.Categories.Any(n => n.Name.ToUpper() == filterCategory.ToUpper()));
```

Comment:
instead of
`n.Name.ToUpper() == filterCategory.ToUpper()`
use 
`string.Equals(n.Name, filterCategory, StringComparison.InvariantCultureIgnoreCase)`